Model/Product/Index/IndexRepository.php

```diff
@@ -99,6 +99,29 @@ public function getById($id)
         return $collection->getFirstItem(); // @codingStandardsIgnoreLine
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function getTotalOutOfSync()
```

Comment:
Add optional Store parameters. If Store is present limit the results only for that store.